core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -524,49 +524,26 @@ private[master] class Master(
   }
 
   /**
-   * Can an app use the given worker? True if the worker has enough memory and we haven't already
-   * launched an executor for the app on it (right now the standalone backend doesn't like having
-   * two executors on the same worker).
+   * Can an app use the given worker?
    */
   private def canUse(app: ApplicationInfo, worker: WorkerInfo): Boolean = {
-    worker.memoryFree >= app.desc.memoryPerSlave && !worker.hasExecutor(app)
+    val enoughResources = worker.memoryFree >= app.desc.memoryPerExecutorMB && worker.coresFree > 0
+    val allowToExecute = app.desc.coresPerExecutor.isDefined || !worker.hasExecutor(app)
```

Comment:
actually, let's just remove this method. We only use it in 1 place and it's small enough that we really don't need it.